web/forms.py

```
@@ -1369,6 +1370,24 @@ def clean_deadline(self):
+        fields = ["completion_image", "completion_link", "completion_notes"]
+        widgets = {
+            "completion_notes": forms.Textarea(attrs={"rows": 3}),
+        }
+
+    def clean(self):
+        cleaned_data = super().clean()
+        image = cleaned_data.get("completion_image")
+        link = cleaned_data.get("completion_link")
+        notes = cleaned_data.get("completion_notes")
+        if not image and not link and not notes:
+            raise forms.ValidationError("Please provide at least one form of proof (image, link, or notes).")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)